dashboard.el

```diff
@@ -388,66 +388,52 @@ Optional argument ARGS adviced function arguments."
-      (setq dashboard-banner-length (window-width)
-            dashboard-buffer-last-width dashboard-banner-length)
-      (with-current-buffer (get-buffer-create dashboard-buffer-name)
-        (let (buffer-read-only)
-          (erase-buffer)
-          (dashboard-insert-banner)
-          (dashboard-insert-page-break)
-          (setq dashboard--section-starts nil)
-          (mapc (lambda (els)
-                  (let* ((el (or (car-safe els) els))
-                         (list-size
-                          (or (cdr-safe els)
-                              dashboard-items-default-length))
-                         (item-generator
-                          (cdr-safe (assoc el dashboard-item-generators))))
-                    (add-to-list 'dashboard--section-starts (point))
-                    (funcall item-generator list-size)
-                    (when recentf-is-on
-                      (setq recentf-list origial-recentf-list))
-                    (setq max-line-length
-                          (max max-line-length (dashboard-maximum-section-length)))
-                    (dashboard-insert-page-break)))
-                dashboard-items)
-          (when dashboard-center-content
-            (when dashboard--section-starts
-              (goto-char (car (last dashboard--section-starts))))
-            (let ((margin (floor (/ (max (- (window-width) max-line-length) 0) 2))))
-              (while (not (eobp))
-                (unless (string-suffix-p (thing-at-point 'line) dashboard-page-separator)
-                  (insert (make-string margin ?\ )))
-                (forward-line 1))))
-          (dashboard-insert-footer))
-        (goto-char (point-min))
-        (dashboard-mode)))
-    (when recentf-is-on
-      (setq recentf-list origial-recentf-list))))
-
-(add-hook 'window-setup-hook
-          (lambda ()
-            ;; 100 means `dashboard-resize-on-hook' will run last
-            (add-hook 'window-size-change-functions 'dashboard-resize-on-hook 100)
-            (dashboard-resize-on-hook)))
+    (prog1
+        (with-current-buffer (get-buffer-create dashboard-buffer-name)
+          (when (or dashboard-force-refresh (not (eq major-mode 'dashboard-mode)))
+            (let (buffer-read-only)
+              (erase-buffer)
+              (dashboard-insert-banner)
+              (setq dashboard--section-starts nil)
+              (mapc (lambda (els)
+                      (let* ((el (or (car-safe els) els))
+                             (list-size
+                              (or (cdr-safe els)
+                                  dashboard-items-default-length))
+                             (item-generator
+                              (cdr-safe (assoc el dashboard-item-generators))))
+                        (push (point) dashboard--section-starts)
+                        (funcall item-generator list-size)
+                        (goto-char (point-max))
+                        (when recentf-is-on
```

Comment:
I think we don't need to restore the `recentf-list` to it's original value (also I think there is a typo to fix in `origial-recentf-list`) It's done at the end of the function.

Reply:
I'm preserving the current behavior here. I _believe_ this exists so that if some section (a) modifies the recentf list, some later section (b) won't be affected. E.g., if the recent files section comes after the org agenda section.

Reply:
Ok, we can leave it that way and maybe later change the recent files behavior in a way that `dashboard-insert-startupify-lists` doesn't know anything about recent files.

I didn't think about each section not been affected, I just thought the "origial" `recentf-list` is saved before any dashboard section consume a file and polluted that list. After all the dashboard is done, we set back the `recentf-list` with  the "origial" list value like nothing happened.